TGUI Crew Manifest

nauticall opened the pull request:
# About the pull request

Modifies the Crew Manifest from its current iteration to a shiny, dynamically-updating TGUI version. Squads and departments are color coded, leaders are highlighted, and status is represented with a marker that displays status when moused-over (e.g. Active, Inactive or Deceased). Search functionality is also included for both name and rank.

Department and role order should remain mostly the same way it is currently, with more senior roles being placed higher up the list of their department, albeit with some minor variations.

Style and backend shamelessly ripped and cannibalized from Aurora. 

Will definitely need testing. This toys with squad assignment stuff a little bit (how, I'm not entirely sure, it just does that) and some things might behave oddly.

# Explain why it's good for the game

Who doesn't like sleek things that do stuff better? This new manifest is prettier, refreshes automatically, and uses TGUI, which is nice.

# Testing Photographs and Procedure
![image](https://github.com/user-attachments/assets/9aef7bf2-9298-47fb-b625-84100d1e4970)
![image](https://github.com/user-attachments/assets/5bb4c443-b703-4aec-93c3-879567146789)
![image](https://github.com/user-attachments/assets/bb425e3e-550e-4604-9180-36f9f2cccef5)


# Changelog
:cl:
ui: Changed the crew manifest to TGUI.
/:cl:

realforest2001 commented:
Needs to account for specialist assignments 
![image](https://github.com/user-attachments/assets/5720977f-02b1-4af8-adae-6288142ea9d5)

zzzmike commented:
Currently, there's no scroll bar on the right, and you also can't middle click to bring up the scroll icon. This makes it a lot more difficult to scroll if your mouse wheel is finnicky, would it be possible to add one or both of these?

nauticall commented:
![image](https://github.com/user-attachments/assets/cf767f5b-f139-4f1e-a57e-c38de9124b59)
![image](https://github.com/user-attachments/assets/d08be9f7-a8fd-42c0-bf3e-a1a0ebbff7ae)
![image](https://github.com/user-attachments/assets/144f7062-af77-41c9-9809-44cf73778c8a)

fixed the spec stuff, and added both scroll bar + search bar functionality :3

nauticall commented:
![image](https://github.com/user-attachments/assets/ac0060dd-9db7-4659-a5ac-2d684d959d9b)

harryob commented on code/datums/crew_manifest.dm:
```diff
@@ -0,0 +1,95 @@
+				break
+
+		if(entry_dept)
+			if(!data[entry_dept])
+				data[entry_dept] = list()
+			data[entry_dept] += list(entry)
```
```suggestion
			LAZYADD(data[entry_dept], list(entry))
```

harryob commented on code/datums/crew_manifest.dm:
```diff
@@ -0,0 +1,95 @@
+				data[entry_dept] = list()
+			data[entry_dept] += list(entry)
+		else
+			if(!data["Miscellaneous"])
+				data["Miscellaneous"] = list()
+			data["Miscellaneous"] += list(entry)
```
ditto

harryob reviewed (changes requested):
nitpicks

harryob commented on code/datums/crew_manifest.dm:
```diff
@@ -0,0 +1,95 @@
+				break
+
+		if(entry_dept)
+			if(!data[entry_dept])
+				data[entry_dept] = list()
```
```suggestion
```
this is what the lazyadd does

harryob commented on code/datums/crew_manifest.dm:
```diff
@@ -0,0 +1,95 @@
+				data[entry_dept] = list()
+			LAZYADD(data[entry_dept], list(entry))
+		else
+			if(!data["Miscellaneous"])
+				data["Miscellaneous"] = list()
```
```suggestion
```
